src/main/java/net/bramp/ffmpeg/builder/FFmpegBuilder.java

```diff
@@ -263,6 +263,21 @@ public FFmpegOutputBuilder addOutput(URI uri) {
     return output;
   }
 
+  /**
+   * Adds new HLS(Http Live Streaming) output file.
+   * <br>
+   * ex) new FFmpegBuilder().addHlsOutput("out.m3u8")
+   *
+   * @param filename output file path
+   *
+   * @return A new {@link FFmpegHlsOutputBuilder}
+   */
+  public FFmpegHlsOutputBuilder addHlsOutput(String filename) {
+    FFmpegHlsOutputBuilder output = new FFmpegHlsOutputBuilder(this, filename);
+    outputs.add(output);
+    return output;
+  }
```

Comment:
I'm unsure about this pattern here. This would be the first time we have a different kind of output builder.

Previously, we have just added properties to the FFmpegOutputBuilder, but I guess what we are implying here is that all the HLS properties are only used for HLS streams, so they shouldn't be on the generic FFmpegOutputBuilder.. If that's the case I guess this pattern is ok.

Reply:
I also thought a lot about this part.

The arguments I added to the `FFmpeglsOutputBuilder` are options that are only used in the hls format, so it doesn't seem appropriate for me to go into the `FFmpegOutputBuilder` either.

So the concern is where are the properties like `hls_time` located.

I think it will be too vast to put in the `FFmpegOutputBuilder`, so I'm thinking of override the `build()` by putting it in the `FFmpegHlsOutputBuilder`. 

Do you have any recommendations for this?

Reply:
Yeah, I think creating a new `FFmpegHlsOutputBuilder` is the way to go, and any hls specific properties should be on it.

The yes overriding build seem appropriate, calling super.build and extending it.

This also makes me wonder if there are other OutputBuilders we should be creating

Reply:
I agree with the statements here. HLS options are pointless for a regular output, so creating a new OutputBuilder is appropriate. @bramp Technically, yes, there are multiple OutputBuilders we could think about. The one most frequently requested would be writing to a Java `InputStream`.